### ISSUE TITLE
Add tutorial for Navigating While Mapping (SLAM)

### ISSUE THREAD

**Issue by abhinavsagar** [turtlebot/turtlebot4#242]
It would be helpful to have a tutorial for Navigating While Mapping (SLAM) like the one on the nav2 documentation using turtlebot3 [https://navigation.ros.org/tutorials/docs/navigation2_with_slam.html](https://navigation.ros.org/tutorials/docs/navigation2_with_slam.html).

**Comment by MikeDegany** [comment]
I did something similar and it worked: 

Run Nav2: 
```ros2 launch turtlebot4_navigation nav2.launch.py namespace:=/<namespace1>```

Run SLAM: 
```ros2 launch turtlebot4_navigation slam.launch.py namespace:=/<namespace1>```

See the map on Rviz: 
```ros2 launch turtlebot4_viz view_robot.launch.py namespace:=/<namespace1>```

**Comment by hilary-luo** [comment]
I agree that this is a nice to have. If someone wants to take this on feel free to create a PR on https://github.com/turtlebot/turtlebot4-user-manual, following the format of the existing tutorials. At this time it isn't a planned priority.

**Comment by halimimaha** [comment]
@MikeDegany @hilary-luo  please i have a question, those steps is for one robot, if i have 2 or 3 robots how can i launch it ? 

thanks in advance

**Comment by hilary-luo** [comment]
@halimimaha Your multiple robots need to be namespaced separately and therefore you will run each launch file required for navigation for each namespace for each of your robots. That being said, it is a very uncommon use case to need to actively map with multiple robots at the same time so I would suggest making sure that you understand what each of these launch files does and that it lines up with what you are trying to do.

**Comment by halimimaha** [comment]
@hilary-luo I already have namespace the robots but the problem is that the ros domain id seems saturated, I cannot launch more than one robot in the same ros domain id :/

**Comment by hilary-luo** [comment]
@halimimaha Please open your own troubleshooting ticket to continue this discussion. I am not sure what you mean by saturated (what symptoms you are seeing) and would need to further understand the details to assist.

**Comment by ajjaya18** [comment]
Hi, can you Help me with turtlebot4  with generating a map?

**Comment by ajjaya18** [comment]
"When I try to map in the world 'depot', my Turtlebot 4 doesn't recognize the objects in the world, but I don't understand why this is happening because in the tutorial, there's already a pre-made map of the 'depot' world, and I want to do the same, but it's not working. Any ideas why this is happening?"

**Comment by ajjaya18** [comment]
@hilary-luo

**Comment by hilary-luo** [comment]
@ajjaya18 Please open your own ticket with all of the details about your system / setup. The info you have provided here isn't enough to really help you off of. Include the system info that it asks for as well as the specific commands you are running, the terminal outputs you get and images of what you are seeing in rviz if it is relevant. I assume that you are already going through the [tutorial for generating a map](https://turtlebot.github.io/turtlebot4-user-manual/tutorials/generate_map.html).

**Comment by ajjaya18** [comment]
![Captura de pantalla de 2024-03-07 11-23-03](https://github.com/turtlebot/turtlebot4/assets/93724428/48d82598-b44f-41e6-8a74-225ec63bfe04)
![Captura de pantalla de 2024-03-07 11-22-41](https://github.com/turtlebot/turtlebot4/assets/93724428/fb439c08-f3c6-4685-8d39-21d656164219)
![Captura de pantalla de 2024-03-07 11-22-30](https://github.com/turtlebot/turtlebot4/assets/93724428/45c3f4c4-c703-4856-9d76-df85eb71252a)
![Captura de pantalla de 2024-03-07 11-22-20](https://github.com/turtlebot/turtlebot4/assets/93724428/e63f6a35-2a21-414e-9920-9f560184c73e)
![Captura de pantalla de 2024-03-07 11-22-19](https://github.com/turtlebot/turtlebot4/assets/93724428/9aaaf43d-c197-4cb6-b886-e10aad9b57bc)
![Captura de pantalla de 2024-03-07 11-22-17](https://github.com/turtlebot/turtlebot4/assets/93724428/2746b164-126a-4a89-8921-d00d7b7d9478)
![Captura de pantalla de 2024-03-07 11-21-16](https://github.com/turtlebot/turtlebot4/assets/93724428/e8bcabfd-8564-4976-bfbc-3ddf04362c5b)
![Captura de pantalla de 2024-03-07 11-20-46](https://github.com/turtlebot/turtlebot4/assets/93724428/2a53448c-cb04-45fb-aae9-ceb569e949b4)
![Captura de pantalla de 2024-03-07 11-20-39](https://github.com/turtlebot/turtlebot4/assets/93724428/34a22852-1daf-46ca-9615-d02558913718)
![Captura de pantalla de 2024-03-07 11-20-38](https://github.com/turtlebot/turtlebot4/assets/93724428/420599a7-fb50-41cb-a4e0-d54074a79cb2)

**Comment by ajjaya18** [comment]
ros2 launch turtlebot4_viz view_robot.launch.py, I run this command in the terminal 
![Captura de pantalla de 2024-03-07 11-22-30](https://github.com/turtlebot/turtlebot4/assets/93724428/0a9015c5-ec1f-42fe-827e-f8ddb093577a)
"Another thing, as you can see in the images, my robot doesn't recognize my depot world items, and I don't understand what I'm missing because there is already a depot map in the files that are in the tutorial. If you can help me, it would be of great help because also, look in rviz, it doesn't create anything on the map, and then I don't know what the error could be."

**Comment by ajjaya18** [comment]
@hilary-luo help me, please

**Comment by ajjaya18** [comment]
I use ros galactic

**Comment by hilary-luo** [comment]
@ajjaya18 please open your own ticket and fill the information out there.
